client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js

```diff
@@ -239,8 +238,7 @@ var FolderToolbarView = Backbone.View.extend({
     },
 
     createNewHistory: function(new_history_name) {
-        var promise = $.post(`${getAppRoot()}api/histories`, { name: new_history_name });
-        return promise;
+        return $.post(`${getAppRoot()}api/histories`, { name: new_history_name });
```

Comment:
when removing the variable usage, maybe it would be worth to add a little comment on what i being returned by the function since it becomes a bit less clear

Reply:
done! thanks